app/src/main/java/com/example/trackflix/fragments/list/ListFragment.kt

```diff
@@ -0,0 +1,124 @@
+import android.view.LayoutInflater
+import android.view.Menu
+import android.view.MenuInflater
+import android.view.MenuItem
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProvider
+import androidx.navigation.fragment.findNavController
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.trackflix.R
+import com.example.trackflix.viewModel.TrackableViewModel
+import com.example.trackflix.databinding.FragmentListBinding
+
+//// TODO: Rename parameter arguments, choose names that match
+//// the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
+//private const val ARG_PARAM1 = "param1"
+//private const val ARG_PARAM2 = "param2"
+
+/**
+ * A simple [Fragment] subclass.
+ * Use the [ListFragment.newInstance] factory method to
+ * create an instance of this fragment.
+ */
+class ListFragment : Fragment() {
+//    // TODO: Rename and change types of parameters
+//    private var param1: String? = null
+//    private var param2: String? = null
+//
+//    override fun onCreate(savedInstanceState: Bundle?) {
+//        super.onCreate(savedInstanceState)
+//        arguments?.let {
+//            param1 = it.getString(ARG_PARAM1)
+//            param2 = it.getString(ARG_PARAM2)
+//        }
+//    }
+
+    private lateinit var binding: FragmentListBinding
+    private lateinit var myTrackableViewModel: TrackableViewModel
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        // Inflate the layout for this fragment
+        //val view = inflater.inflate(R.layout.fragment_list, container, false)
+        //return view
+
+        binding = FragmentListBinding.inflate(layoutInflater, container, false)
+        val view = binding.root
+
+
+        val adapter = ListAdapter()
+        val recyclerView = binding.recyclerView
+        recyclerView.adapter = adapter
+        recyclerView.layoutManager = LinearLayoutManager(requireContext())
+
+        myTrackableViewModel = ViewModelProvider(this).get(TrackableViewModel::class.java)
+        myTrackableViewModel.readAllData.observe(viewLifecycleOwner, Observer {trackable ->
+            adapter.setData(trackable)
+        })
+
+        binding.floatingActionButton.setOnClickListener{
+            findNavController().navigate(R.id.action_listFragment_to_addFragment)
+        }
+
+        //add menu
+        setHasOptionsMenu(true)
+
+        return view
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        inflater.inflate(R.menu.delete_menu, menu)
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        if(item.itemId == R.id.menu_delete){
+            deleteAllTrackables()
+        }
+        return super.onOptionsItemSelected(item)
+    }
+
+    private fun deleteAllTrackables() {
```

Comment:
This is quite a scary option. We should design the UI so that it isn't easily accessible.